service/service.go

```diff
@@ -151,6 +157,9 @@ func (s *Service) closeStatus(statuses chan DeployStatus) {
 
 // getDependency returns dependency dependencyKey or a not found error.
 func (s *Service) getDependency(dependencyKey string) (*Dependency, error) {
+	if dependencyKey == MainServiceKey {
```

Comment:
this pr is about moving service configuration out of dependencies. it doesn't make much sense this func to return the service configuration

Reply:
this simplifies a lot the `extractVolumesFrom` function. I agree it's not ideal but will add quite a lot of complexity otherwise

Reply:
also this ensures that previous services (the one with still the service in dependencies) can run

Reply:
this logic has been moved to the volumefrom